crates/miden-lib/asm/kernels/transaction/lib/account.masm

```diff
@@ -529,22 +549,67 @@ end
 #! Panics if:
 #! - the requested storage slot type is not map.
 export.get_map_item
-    # get the storage slot type
-    dup exec.get_storage_slot_type
-    # => [slot_type, index, KEY]
-
-    # check if storage slot type is map
-    exec.constants::get_storage_slot_type_map eq
-    assert.err=ERR_ACCOUNT_READING_MAP_VALUE_FROM_NON_MAP_SLOT
-    # => [index, KEY]
-
+    # duplicate index for later use
     dup movdn.5
     # => [index, KEY, index]
 
     # fetch the account storage item, which is ROOT of the map
     exec.get_item swapw
     # => [KEY, ROOT, index]
 
+    exec.get_map_item_raw
+end
+
+#! Returns the VALUE located under the specified KEY within the map contained in the given
+#! account storage slot at its initial state (beginning of transaction).
+#!
+#! Inputs:  [index, KEY]
+#! Outputs: [INIT_VALUE]
+#!
+#! Note:
+#! - We assume that index has been validated and is within bounds.
```

Comment:
Is this assumption correct? I couldn't find any place that validates the index

Reply:
The index is validated in `api.masm`, i.e. in `account_get_item` or `account_get_item_init` when calling into `apply_storage_offset`, so the assumption in the account module is fine. Analogously for the map item APIs.

Also note that this will soon change with named storage slots, so we can reconsider the validation approach there if desired.